packages/react/src/analytics/integrations/Vitorino/Vitorino.ts

```diff
@@ -72,204 +67,87 @@ export default class Vitorino extends integrations.Integration<VitorinoIntegrati
-      this.initializePromiseResolve = null;
-    }
-  };
-
-  /**
-   * Builds the object needed for the Vitorino pixel configuration with all necessary
-   * data.
-   *
-   * @param data - Analytics' track data.
-   *
-   * @returns - The payload to be sent to the pixel.
-   */
-  buildConfigData(data: SetUserEventData): Config {
-    const dataWithWebEventType = data as SetUserEventData & {
-      context: WebContextType;
-    };
-    const correlationId = data.user.localId;
-    const { tenantId, clientId, web } = dataWithWebEventType.context;
-    const origin = web?.window?.location?.origin;
-
-    if (!tenantId || !clientId) {
-      throw new Error(MISSING_CONTEXT_ERROR_MESSAGE);
-    }
-
-    return {
-      correlationId,
-      tenantId,
-      clientId,
-      origin,
-      customerId: utils.getCustomerIdFromUser(data.user),
-      environment: this.getEnvironmentFromOptions(this.options),
-      fields: {
-        sensitiveFields: this.options?.sensitiveFields || [],
-        secretFields: this.options?.secretFields || [],
-      },
-      network: this.options.network || {
-        proxy: origin,
-        path: `${MARKETING_API_PREFIX}${postTracking.POST_TRACKINGS_PATHNAME}`,
-      },
-    };
+  static override createInstance<Options extends VitorinoIntegrationOptions>(
+    this: IntegrationFactory<Options>,
+    options: Options,
+    loadData: LoadIntegrationEventData,
+    analytics: StrippedDownAnalytics,
+  ) {
+    return new this(options, loadData, analytics);
   }
 
   /**
-   * Returns the correspondent Vitorino page type for the event.
-   *
-   * @param event - The analytics event to filter by.
+   * This integration is required, so it must be loaded independently of user
+   * consent.
    *
-   * @returns - The correspondent Vitorino's page view(s).
+   * @returns If the integration should load or not.
    */
-  getPageTypeFromEvent(event: EventData<TrackTypesValues>['event']) {
-    const mapper = this.getEventsMapper();
-
-    if (!mapper || typeof mapper !== 'object') {
-      throw new Error(INVALID_MAPPER_RETURN_TYPE_ERROR);
-    }
-
-    return mapper[event];
+  static override shouldLoad() {
+    return true;
   }
 
   /**
-   * Calculates the correct environment to pass to the Vitorino configuration.
-   *
-   * @param options - Integration options.
+   * When the user ID is available, config the vitorino and resolve the Promise so
+   * the tracking can begin.
    *
-   * @returns The correct environment.
+   * @param data - Analytics' event data.
    */
-  getEnvironmentFromOptions(options: VitorinoIntegrationOptions) {
-    const { environment } = options;
-    const safeEnvironment =
-      environment || window?.__BUILD_CONTEXT__?.env?.NODE_ENV;
-
-    return safeEnvironment === ENVIRONMENT_TYPES.prod
-      ? window.Vitorino.Environment.PRODUCTION
-      : window.Vitorino.Environment.SANDBOX;
+  override async onSetUser(data: SetUserEventData): Promise<void> {
+    await this.forter?.onSetUser(data);
```

Comment:
Can this be done in parallel instead of calling forter's onSetUser first and only then calling riskified's onSetUser? Also, do they need to be awaited? If so, and they can be done in parallel, use `Promise.all()` instead.

Reply:
Fixed.